### ISSUE TITLE
Write standardadized Machine Learning model training / evaluation file

### ISSUE THREAD

**Issue by timonmerk** [neuromodulation/py_neuromodulation#11]
- non shuffled cross validation. Maybe grouped k fold cross validation also makes sense here https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GroupKFold.html 
- Support LDA, SVM, NN, XGBOOST, RF 
- Provide Bayesian Optimization https://scikit-optimize.github.io/stable/ framework for hyperparameter optimization
